pkg/controller/currency-controller.go

```diff
@@ -3,22 +3,23 @@ package controller
 import (
 	"net/http"
 
-	"genesis-currency-api/internal/service"
+	"genesis-currency-api/pkg/interface/service"
+
 	"github.com/gin-gonic/gin"
 )
 
-type CurrencyController struct {
-	currencyService *service.CurrencyService
+type CurrencyControllerImpl struct {
+	currencyService service.CurrencyService
 }
 
-func (c *CurrencyController) GetLatest(ctx *gin.Context) {
+func (c *CurrencyControllerImpl) GetLatest(ctx *gin.Context) {
 	result := c.currencyService.GetCurrencyRate()
 	ctx.String(http.StatusOK, "%f", result.Number)
 }
 
-// CurrencyRegisterRoutes creates an instance of CurrencyController and registers routes for it.
-func CurrencyRegisterRoutes(r *gin.Engine, s *service.CurrencyService) {
-	c := &CurrencyController{
+// RegisterCurrencyRoutes creates an instance of CurrencyControllerImpl and registers routes for it.
+func RegisterCurrencyRoutes(r *gin.Engine, s service.CurrencyService) {
```

Comment:
Look like this func also violates `S`. 

Is create an instance and register routes. 

Also, I prefere to move this to another package (api) in internal directory. Create constructor to all handlers (controllers) 

In out projects we use struct like this
<img width="292" alt="image" src="https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-AlwaysSayNo/assets/25195930/031a5ba9-e7b3-4f01-a98d-b57182071b71">

Reply:
Done.